### PR TITLE
[libxml2] Add link libraries for target LibXml2::LibXml2

### LINKED ISSUE #14468: libxml2 vcpkg-cmake-wrapper.cmake bugs
**Describe the bug**
In file vcpkg-cmake-wrapper.cmake of package LibXml2, the target LibXml2::LibXml2 is not linked against all necessary libraries

**Environment**
- OS: Windows

**To Reproduce**
1. ./vcpkg install libxml2
2. find_package(LibXml2)
target_link_libraries(main PRIVATE LibXml2::LibXml2)
3. ZLIB_LIBRARIES and LIBLZMA_LIBRARIES are not linked, neither ws2_32 on windows

**Expected behavior**
LibXml2::LibXml2 target should link with Iconv, Charset, lzma, zlib, ws2_32 (windows only) libraries

### PR DESCRIPTION
**Pull request by LilyWangL**
**Describe the pull request**

- What does your PR fix? Fixes #14468
Target `LibXml2::LibXml2` is not linked against all necessary libraries in **vcpkg-cmake-wrapper.cmake**. Target `LibXml2::LibXml2` should link with `Iconv`, `Charset`, `lzma`, `zlib`, `ws2_32` (windows only) libraries.

### CODE REVIEW

**Review by JackBoosY — APPROVED** [review]

**Review by jmelas — CHANGES_REQUESTED** [review]
I think still LIBXML2_LIBRARIES is not populated correctly.
The missing libraries should first be added to LIBXML2_LIBRARIES variable
and then target LibXml2::LibXml2 should link with this variable.

**Comment by LilyWangL** [comment]
> I think still LIBXML2_LIBRARIES is not populated correctly.
> The missing libraries should first be added to LIBXML2_LIBRARIES variable
> and then target LibXml2::LibXml2 should link with this variable.

Due to `libiconv` and `libcharset` don't create lib on Linux, so it need not link to them when `CMAKE_SYSTEM_NAME STREQUAL "Linux"`. I think this handle is correct.

**Review by jmelas — CHANGES_REQUESTED** [review]
Now it's broken on macOS.
This is my proposition:

    if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
        list(APPEND LIBXML2_LIBRARIES m)
    else()
        list(APPEND LIBXML2_LIBRARIES ${ICONV_LIBRARIES} ${CHARSET_LIBRARIES} ${LIBLZMA_LIBRARIES} ${ZLIB_LIBRARIES})
        if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
            list(APPEND LIBXML2_LIBRARIES ws2_32)
        endif()
    endif()
    if(TARGET LibXml2::LibXml2)
        target_link_libraries(LibXml2::LibXml2 INTERFACE ${LIBXML2_LIBRARIES})
    endif()

**Comment by LilyWangL** [comment]
Due to `libiconv` and `libcharset` don't create lib on MACOS, I think this shoule be changed to:
```
    list(APPEND LIBXML2_LIBRARIES ${LIBLZMA_LIBRARIES} ${ZLIB_LIBRARIES})
    if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
        list(APPEND LIBXML2_LIBRARIES m)
    elseif(CMAKE_SYSTEM_NAME STREQUAL "Windows")
        list(APPEND LIBXML2_LIBRARIES ${ICONV_LIBRARIES} ${CHARSET_LIBRARIES} ws2_32)
	endif()
	if(TARGET LibXml2::LibXml2)
        target_link_libraries(LibXml2::LibXml2 INTERFACE ${LIBXML2_LIBRARIES})
    endif()
```

**Review by jmelas — APPROVED** [review]

**Review by jmelas — APPROVED** [review]

File: ports/libxml2/vcpkg-cmake-wrapper.cmake [hunk 1]
```diff
@@ -16,11 +16,11 @@ if(LibXml2_FOUND)
...
-        if(TARGET LibXml2::LibXml2)
-           target_link_libraries(LibXml2::LibXml2 INTERFACE ${ICONV_LIBRARIES} ${CHARSET_LIBRARIES})
+	if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
+            list(APPEND LIBXML2_LIBRARIES ws2_32)
         endif()
     endif()
-    if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
-        list(APPEND LIBXML2_LIBRARIES ws2_32)
+    if(TARGET LibXml2::LibXml2)
+        target_link_libraries(LibXml2::LibXml2 INTERFACE ${LIBXML2_LIBRARIES})
```
Comment (Neumann-A) [inline ports/libxml2/vcpkg-cmake-wrapper.cmake:24]: Simply no. This is linking itself. LIBXML2_LIBRARIES is not empty after the `find_package` call but contains the libxml2 library.

Comment (strega-nil) [reply ports/libxml2/vcpkg-cmake-wrapper.cmake:24]: Yeah, I think I agree with this; could we switch to the old code, but add `target_link_libraries` below the second `list(APPEND`?

**Comment by JackBoosY** [comment]
Replaced by #14588.

**Pull request closed without merging**